dbus_service.py

```diff
@@ -461,23 +468,39 @@ def _update(self):
             pre = "/Ac/" + self.pvinverterphase
 
             if self.is_data_up2date():
-                (power, pvyield, current, voltage) = self.get_values_for_inverter()
+                (power, pvyield, current, voltage, dc_voltage) = self.get_values_for_inverter()
 
                 if self.dry_run:
                     logging.info("DRY RUN. No data is sent!!")
                 else:
-                    self._dbusservice[pre + "/Voltage"] = voltage
-                    self._dbusservice[pre + "/Current"] = current
-                    self._dbusservice[pre + "/Power"] = power
-                    self._dbusservice["/Ac/Power"] = power
-                    if power > 0:
-                        self._dbusservice[pre + "/Energy/Forward"] = pvyield
-                        self._dbusservice["/Ac/Energy/Forward"] = pvyield
-
-                logging.debug("Inverter #%d Power (/Ac/Power): %s", self.pvinverternumber, power)
-                logging.debug("Inverter #%d Energy (/Ac/Energy/Forward): %s", self.pvinverternumber, pvyield)
-                logging.debug("---")
-
+                    if self._servicename == "com.victronenergy.inverter":
```

Comment:
I don't like nested ifs. Maybe extract some method here?
(My vision is to have different handler classes for the different dtu_variants and for different servicenames... But I don't have a clear picture of that...)

Reply:
I moved it to a method. It should be "cleaner" now. I want to refactor the service to introduce a "device-class"-based class like inverter, PVinverter, grid, etc.  This would be my vision but I don't know yet if I have the time for this.

Reply:
Yes, exactly.